src/main/java/io/horizen/lambo/CarRegistryApplicationState.java

```diff
@@ -1,39 +1,85 @@
 package io.horizen.lambo;
 
+import com.google.inject.Inject;
 import com.horizen.block.SidechainBlock;
 import com.horizen.box.Box;
 import com.horizen.proposition.Proposition;
 import com.horizen.state.ApplicationState;
 import com.horizen.state.SidechainStateReader;
 import com.horizen.transaction.BoxTransaction;
+import io.horizen.lambo.car.box.CarBox;
+import io.horizen.lambo.car.box.CarSellOrderBox;
+import io.horizen.lambo.car.services.CarInfoDBService;
+import io.horizen.lambo.car.transaction.CarDeclarationTransaction;
 import scala.util.Success;
 import scala.util.Try;
-
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
+
 
-// There is no custom logic for Car registry State now.
-// TODO: prevent the declaration of CarBoxes which car information already exists in the previously added CarBoxes or CarSellOrderBoxes.
 public class CarRegistryApplicationState implements ApplicationState {
+	
+	private CarInfoDBService carInfoDbService;
+
+	@Inject
+	public CarRegistryApplicationState(CarInfoDBService carInfoDbService) {
+	    this.carInfoDbService = carInfoDbService;
+	}
+	
     @Override
     public boolean validate(SidechainStateReader stateReader, SidechainBlock block) {
-        return true;
+	    return true;
     }
 
     @Override
     public boolean validate(SidechainStateReader stateReader, BoxTransaction<Proposition, Box<Proposition>> transaction) {
-        // TODO: here we expect to go though all CarDeclarationTransactions and verify that each CarBox reflects to unique Car.
+        // we go though all CarDeclarationTransactions and verify that each CarBox reflects to unique Car.
+        if (CarDeclarationTransaction.class.isAssignableFrom(transaction.getClass())){
```

Comment:
why not `if(transaction instanceof CarDeclarationTransaction)` ?

Reply:
I tried it but it gives this compilation error:
java: incompatible types: com.horizen.transaction.BoxTransaction<com.horizen.proposition.Proposition,com.horizen.box.Box<com.horizen.proposition.Proposition>> cannot be converted to io.horizen.lambo.car.transaction.CarDeclarationTransaction

(to be honest I don't understand what is the problem because the hierachy seems compatible!)

Reply:
I got it work by adding explicitly this generic signature to CarDeclarationTransaction
CarDeclarationTransaction<P extends Proposition, B extends  NoncedBox<P>>